src/pathfinder/water_regions.cpp

```diff
@@ -246,10 +246,10 @@ static TileIndex GetEdgeTileCoordinate(int region_x, int region_y, DiagDirection
 static WaterRegion GetUpdatedWaterRegion(uint16_t region_x, uint16_t region_y)
 {
 	const TWaterRegionIndex index = GetWaterRegionIndex(region_x, region_y);
-	WaterRegion water_region(region_x, region_y, _water_region_data[index]);
-	if (!_is_water_region_valid[index]) {
+	WaterRegion water_region(region_x, region_y, _water_region_data[index.base()]);
+	if (!_is_water_region_valid[index.base()]) {
```

Comment:
Maybe these two containers should be wrapped with `ReferenceThroughBaseContainer` to avoid all the extra `.base()` calls?